client/auth/login.jsx

```diff
@@ -135,7 +149,7 @@ module.exports = React.createClass( {
 							<FormTextInput
 								name="login"
 								ref="login"
-								disabled={ requires2fa || inProgress }
+								disabled={ !! requires2fa || inProgress }
```

Comment:
I see what you mean about the oddity of having `!!` next to `||` and I think there's a third way to do this that's better. `requires2fa` by its name indicates a boolean value, "does it require 2FA?" It's _actually_ encoding a required type of 2FA or nothing. How about we fix this? If it's too much to fix through the file (which probably _isn't_ the case) then we could at least solve it locally in `render()`

```js
const { requires2fa: required2faType } = this.state;
const requires2fa = !! required2faType;

disabled={ requires2fa || inProgress }
```